### PR TITLE
Add doc string for type docs

### PR DESCRIPTION
**Pull request by Synicix**
Finished with the doc string for now. Will work on rst and github build action on Monday to deploy this.
- Refactor a bunch of code to remove usage of any and make the code slightly more consistent
- Added typedoc doc string to all classes and functions
- Refactoring of props params for react components

### CODE REVIEW

**Comment by Synicix** [comment]
Will add change log in a bit

**Review by mahos — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
 ## [Unreleased]
+## Added
+- Docstrings for all classes and functions (#112) PR #124
+
+## Changed
+- Replaced usage of any with more strongly typed data types (#125) PR #124
+- Replaced the direct react component props definition with interface PR #124
+- Replaced state and props with interface instead of types PR #124
+
+## [0.1.0-beta.2]
```
Comment (guzman-raphael) [inline CHANGELOG.md]: There is a merge conflict here. Just need to pull from latest in `dev` to resolve this.

File: src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx [hunk 2]
```diff
@@ -46,9 +63,14 @@ class SecondaryTableAttribute extends TableAttribute {
...
 
-  static getAttributeInputBlock(secondaryTableAttribute: SecondaryTableAttribute, currentValue: any, handleChange: any) {
+  /**
+   * Construct input block based on secondaryTableAttribute by extracting it type and other information
+   * @param secondaryTableAttribute 
+   * @param currentValue urrentValue of the input block for binding
+   * @param handleChange Call back function for when the value of the input block changes
+   * @returns 
+   */
+  static getSecondaryAttributeInputBlock(secondaryTableAttribute: SecondaryTableAttribute, currentValue: any, handleChange: (event: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>, attributeName: string) => void) {
```
Comment (guzman-raphael) [inline src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx:73]: Here is one `any` that was missed.

File: src/Components/MainTableView/DeleteTuple/DeleteTuple.tsx [hunk 3]
```diff
@@ -3,32 +3,32 @@ import "./DeleteTuple.css";
...
-  deleteStatusMessage: string, // for GUI to show
-  isGettingDependencies: boolean, // for loading animation status
-  deleteAccessible: boolean // valdiation result of accessibility from check dependency component
+
+interface DeleteTupleProps {
+  token: string;
+  selectedSchemaName: string;
+  selectedTableName: string;
+  tableAttributesInfo?: TableAttributesInfo;
+  selectedTableEntry?: any;
```
Comment (guzman-raphael) [inline src/Components/MainTableView/DeleteTuple/DeleteTuple.tsx:15]: `any` use

File: src/Components/MainTableView/Filter/FilterCard.tsx [hunk 4]
```diff
@@ -8,14 +8,22 @@ import TableAttributeType from '../enums/TableAttributeType';
 
 import './FilterCard.css'
 
+interface FilterCardProps {
+  index: number;
+  restriction: Restriction;
+  tableAttributes: Array<TableAttribute>;
+  updateRestriction: any;
+  deleteFilterCard: any;
```
Comment (guzman-raphael) [inline src/Components/MainTableView/Filter/FilterCard.tsx:16]: `any` use

File: src/Components/MainTableView/InsertTuple/InsertTuple.tsx [hunk 5]
```diff
@@ -1,41 +1,34 @@
...
+  selectedTableName: string;
+  tableAttributesInfo?: TableAttributesInfo;
+  selectedTableEntry?: any; // Tuple that that is checked
+  fetchTableContent: () => void;
+  clearTupleSelection: () => void;
+  insertInAction: (isWaiting: boolean) => void; // for loading/waiting animation while insert takes place
+}
+
+interface InsertTupleState {
+  tupleBuffer: any; // Tuple buffer to stored the values typed in by the user
```
Comment (guzman-raphael) [inline src/Components/MainTableView/InsertTuple/InsertTuple.tsx:23]: `any` use

File: src/Components/MainTableView/TableView.tsx [hunk 6]
```diff
@@ -20,24 +20,34 @@ enum CurrentView {
...
+  tableAttributesInfo?: TableAttributesInfo; // TableAttributesInfo object that stores all info on current table attirbutes
+  currentView: CurrentView; // Switcher between Content and Info view
+  tableContentNeedRefresh: boolean; // Boolean trigger if the tableContent needs to be refreshed
+  tableDefinitionNeedRefresh: boolean; // Boolean tirgger if the table Definition needs to be refreshed
+  numberOfTuplesPerPage: number; // Number of tuples to view per page
+  setNumberOFTuplesPerPageTimeout: ReturnType<typeof setTimeout>; // Timeout for when to actaully apply the change in numberOfTuples per page
+  totalNumOfTuples: number; // Total number of tuples for the given table
+  currentPageNumber: number; // Current page number that is being rendered
+  maxPageNumber: number; // The max page number which is computed upon fetching the table tuples
+  tableContentData: Array<any>; // The tuples of the table stored in an array
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableView.tsx:40]: `any` use

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx [hunk 7]
```diff
@@ -3,35 +3,36 @@ import TableAttribute from '../DataStorageClasses/TableAttribute';
...
-  tupleBuffer: any, // Object to stored the values typed in by the user
-  errorMessage: string, // Error message string for failed inserts
-  dependencies: Array<any>, // list of dependencies pushed from checkDependency Component
-  updateAccessible: boolean // disables submit button if any of the dependencies are inaccessible
+interface UpdateTupleProps {
+  token: string;
+  selectedSchemaName:string;
+  selectedTableName: string;
+  tableAttributesInfo?: TableAttributesInfo;
+  selectedTableEntry: any; // Tuple that that is checked
```
Comment (guzman-raphael) [inline src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:15]: `any` use

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx [hunk 8]
```diff
@@ -3,35 +3,36 @@ import TableAttribute from '../DataStorageClasses/TableAttribute';
...
-    selectedTableName: string, 
-    tableAttributesInfo?: TableAttributesInfo, 
-    fetchTableContent: any, 
-    clearEntrySelection: any,
-    selectedTableEntry: any,
-    updateInAction: any // for loading animation status
-  }, 
-  updateTupleState> {
+interface UpdateTupleState {
+  tupleBuffer: any; // Tuple buffer to stored the values typed in by the user
```
Comment (guzman-raphael) [inline src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:22]: `any` use

File: src/Components/SideMenu/SideMenu.tsx [hunk 9]
```diff
@@ -18,7 +18,7 @@ type HomeSideMenuState = {
  * SideMenu component that handles listing schemas and tables
  * 
  */
-class SideMenu extends React.Component<{token: string, selectedSchema: string, selectedTableName: string, selectedTableType: TableType, handleTableSelection: any}, HomeSideMenuState> {
+export default class SideMenu extends React.Component<{token: string, selectedSchema: string, selectedTableName: string, selectedTableType: TableType, handleTableSelection: any}, HomeSideMenuState> {
   constructor(props: any) {
```
Comment (guzman-raphael) [inline src/Components/SideMenu/SideMenu.tsx:22]: `any` use

File: src/Components/SideMenu/TableList.tsx [hunk 10]
```diff
@@ -39,31 +50,40 @@ class ParentTableListEntry extends TableListEntry {
...
-  viewAllPartTables: boolean,
-  tablesToSort: any,
-  hidePartTable: Array<string>,
-  tableList: Array<ParentTableListEntry>,
-  restrictedTableList: Array<ParentTableListEntry>,
-  searchString: string,
-  currentTableSortMode: TableSortMode,
+interface TableListProps {
+  token: string;
+  tableListDict: any;
```
Comment (guzman-raphael) [inline src/Components/SideMenu/TableList.tsx:64]: `any`

File: src/Components/SideMenu/TableList.tsx [hunk 11]
```diff
@@ -39,31 +50,40 @@ class ParentTableListEntry extends TableListEntry {
...
+  selectedTableName: string;
+  selectedTableType: TableType;
+  onTableSelection: any,
```
Comment (guzman-raphael) [inline src/Components/SideMenu/TableList.tsx:67]: `any` use

File: src/Components/MainTableView/DataStorageClasses/PrimaryTableAttribute.tsx [hunk 12]
```diff
@@ -37,13 +53,18 @@ class PrimaryTableAttribute extends TableAttribute {
...
 
-  static getAttributeInputBlock(tableAttribute: PrimaryTableAttribute, currentValue: any, handleChange: any) {
+  /**
+   * Construct input block based on primaryTableAttribute by extracting it type and other information
+   * @param tableAttribute 
+   * @param currentValue 
+   * @param handleChange 
+   * @returns 
+   */
+  static getPrimaryAttributeInputBlock(tableAttribute: PrimaryTableAttribute, currentValue: any, handleChange: (event: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>, attributeName: string) => void) {
```
Comment (guzman-raphael) [inline src/Components/MainTableView/DataStorageClasses/PrimaryTableAttribute.tsx:63]: `any` use

File: src/Components/MainTableView/DataStorageClasses/Restriction.tsx [hunk 13]
```diff
@@ -4,15 +4,14 @@ import TableAttribute from './TableAttribute'
 /**
  * Class to handle the datastorage of restrctions FilterCards
  */
-class Restriction {
+export default class Restriction {
   id: number
   tableAttribute?: TableAttribute
   restrictionType?: RestrictionType
   value?: any
```
Comment (guzman-raphael) [inline src/Components/MainTableView/DataStorageClasses/Restriction.tsx:11]: `any` use

File: src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx [hunk 14]
```diff
@@ -1,362 +1,388 @@
...
-      else if (tableAttribute.attributeType === TableAttributeType.TIME) {
-        typeString = "HH:MM:SS";
+  /**
+   * Helper function to handle the creation of input block based on the corresponding table attribute
+   * @param tableAttribute TableAttribute object to be used for extracting type 
+   * @param currentValue Current value of the input block
+   * @param defaultValue Any default value for input blocks that support it
+   * @param handleChange Call back function for when the user make a change to the input block
+   */
+  static getAttributeInputBlock(tableAttribute: TableAttribute, currentValue: any, defaultValue: string = '', handleChange: (event: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>, attributeName: string) => void) {
```
Comment (guzman-raphael) [inline src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx:211]: `any` use

File: src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx [hunk 15]
```diff
@@ -1,362 +1,388 @@
...
+    else if (tableAttribute.attributeType === TableAttributeType.UUID) {
+      return <input type="text" value={currentValue} defaultValue={defaultValue} id={tableAttribute.attributeName} onChange={(e) => handleChange(e, tableAttribute.attributeName)}></input>
+
+    }
+    else if (tableAttribute.attributeType === TableAttributeType.DATE) {
+      return <input type="date" value={currentValue} defaultValue={defaultValue === '' ? undefined : defaultValue.replaceAll('"', "")} id={tableAttribute.attributeName} onChange={(e) => handleChange(e, tableAttribute.attributeName)}></input>
+    }
+    else if (tableAttribute.attributeType === TableAttributeType.DATETIME || tableAttribute.attributeType === TableAttributeType.TIMESTAMP) {
+      var splitResult = [undefined, undefined]
+      var defaultValueSplitResult: any = ['', '']
```
Comment (guzman-raphael) [inline src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx:337]: `any` use

File: src/Components/MainTableView/InsertTuple/InsertTuple.tsx [hunk 16]
```diff
@@ -1,41 +1,34 @@
...
 
-type insertTupleState = {
-  tupleBuffer: any // Object to stored the values typed in by the user
-  errorMessage: string // Error message string for failed inserts
+interface InsertTupleProps {
+  token: string;
+  selectedSchemaName:string;
+  selectedTableName: string;
+  tableAttributesInfo?: TableAttributesInfo;
+  selectedTableEntry?: any; // Tuple that that is checked
```
Comment (guzman-raphael) [inline src/Components/MainTableView/InsertTuple/InsertTuple.tsx:16]: `any` use

File: src/Components/MainTableView/TableContent.tsx [hunk 17]
```diff
@@ -13,62 +13,50 @@ import TableAttributeType from './enums/TableAttributeType'
...
-  dragStart: number, // part of table column resizer feature
-  dragDistance: number, // part of table column resizer feature
-  resizeIndex: any, // part of table column resizer feature
-  isWaiting: boolean, // tells the UI to display loading icon while insert/update/delete are in action
+interface TableContentProps {
+  token: string;
+  selectedSchemaName: string;
+  selectedTableName: string;
+  selectedTableType: TableType;
+  contentData: Array<any>; // Array of tuples obtain from the fetch of a table
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableContent.tsx:28]: `any` use

File: src/Components/MainTableView/TableContent.tsx [hunk 18]
```diff
@@ -13,62 +13,50 @@ import TableAttributeType from './enums/TableAttributeType'
...
+  currentSelectedTableActionMenu: TableActionType;
+  hideTableActionMenu: boolean;
+  selectedTupleIndex: number; // Index of selected tuple
+  selectedTuple?: {}; // Has to be an object with each attribute name as key cause the way tuple_buffer is handle in the subcomponents
+  showWarning: boolean; // text warning when duplicate selection is made for delete/update, most likely to be take out once disable checkbox feature is finished
+  isDisabledCheckbox: boolean; // tells the UI to disable any other checkboxes once there is already a selection in delete/update mode
+  headerWidth: number; // part of table column resizer feature
+  dragStart: number; // part of table column resizer feature
+  dragDistance: number; // part of table column resizer feature
+  resizeIndex: any; // part of table column resizer feature
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableContent.tsx:50]: `any` use

File: src/Components/MainTableView/TableView.tsx [hunk 19]
```diff
@@ -20,24 +20,34 @@ enum CurrentView {
...
+  errorMessage: string; // Error message buffer
+  isLoading: boolean; // Boolean for loading animation
+  restrictions: Array<Restriction>; // Storage for the last requested restrction to deal with case such as numberOfTuplesPerPage change
+}
+
+/**
+ * Parent component for handling displaying TableContent and TableInfo
+ */
+export default class TableView extends React.Component<TableViewProps, TableViewState> {
   constructor(props: any) {
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableView.tsx:51]: `any` use

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx [hunk 20]
```diff
@@ -3,35 +3,36 @@ import TableAttribute from '../DataStorageClasses/TableAttribute';
...
+  errorMessage: string; // Error message string for failed inserts
+  updateAccessible: boolean; // disables submit button if any of the dependencies are inaccessible
+}
+
+/**
+ * Component for Update tuples in a given table
+ */
+export default class UpdateTuple extends React.Component<UpdateTupleProps, UpdateTupleState> {
   constructor(props: any) {
```
Comment (guzman-raphael) [inline src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:31]: `any` use

File: CHANGELOG.md [hunk 21]
```diff
@@ -2,6 +2,15 @@
 
 Observes [Semantic Versioning](https://semver.org/spec/v2.0.0.html) standard and [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) convention.
 
+## [Unreleased]
+## Added
```
Comment (guzman-raphael) [inline CHANGELOG.md:6]:
```suggestion
### Added
```

File: CHANGELOG.md [hunk 22]
```diff
@@ -2,6 +2,15 @@
...
+- Docstrings for all classes and functions (#112) PR #124
+
+## Changed
```
Comment (guzman-raphael) [inline CHANGELOG.md:9]:
```suggestion
### Changed
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@Synicix Valiant effort here in cleaning things up. :+1: 

I did spot various more instances of `any` here within the review. Probably a could idea to search across the entire directory for all instances (in VSCode: CRTL+SHIFT+F). I did not confirm if there's more but there might be.

File: src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx (hunk 2 above)
Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx:73]: That one I left it as any cause that requires mutiple types and heavily depends on what the input block type is. I could make it more specific but it is quite a bit of types to account for and will probably introduced bugs if I miss one.

File: src/Components/MainTableView/DeleteTuple/DeleteTuple.tsx (hunk 3 above)
Comment (Synicix) [reply src/Components/MainTableView/DeleteTuple/DeleteTuple.tsx:15]: That has to be any cause the object passing back has different attirbutes depending on what table is being selected, thus there is no nice way to restrict the type.

File: src/Components/MainTableView/Filter/FilterCard.tsx (hunk 4 above)
Comment (Synicix) [reply src/Components/MainTableView/Filter/FilterCard.tsx:16]: well, missed that one.

File: src/Components/MainTableView/InsertTuple/InsertTuple.tsx (hunk 5 above)
Comment (Synicix) [reply src/Components/MainTableView/InsertTuple/InsertTuple.tsx:23]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx (hunk 7 above)
Comment (Synicix) [reply src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:15]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx (hunk 8 above)
Comment (Synicix) [reply src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:22]: Same thing as the selectedTaableEntry

File: src/Components/SideMenu/SideMenu.tsx (hunk 9 above)
Comment (Synicix) [reply src/Components/SideMenu/SideMenu.tsx:22]: Refactored

File: src/Components/SideMenu/TableList.tsx (hunk 11 above)
Comment (Synicix) [reply src/Components/SideMenu/TableList.tsx:67]: fixed

File: src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx (hunk 2 above)
Comment (guzman-raphael) [reply src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx:73]: @Synicix I see, fair enough. I would say then we can keep this as `any` but let's add a note to the docs to indicate why `any` was chosen here. In the docs for that argument, you could say something like

`@param currentValue CurrentValue of the input block for binding. Type any used here as there are many possible types with all the available input blocks.`

File: src/Components/SideMenu/TableList.tsx (hunk 10 above)
Comment (Synicix) [reply src/Components/SideMenu/TableList.tsx:64]: Fixed

File: src/Components/MainTableView/DataStorageClasses/Restriction.tsx (hunk 13 above)
Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/Restriction.tsx:11]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx (hunk 14 above)
Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx:211]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/TableContent.tsx (hunk 17 above)
Comment (Synicix) [reply src/Components/MainTableView/TableContent.tsx:28]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/InsertTuple/InsertTuple.tsx (hunk 16 above)
Comment (Synicix) [reply src/Components/MainTableView/InsertTuple/InsertTuple.tsx:16]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/TableContent.tsx (hunk 18 above)
Comment (Synicix) [reply src/Components/MainTableView/TableContent.tsx]: fixed

File: src/Components/MainTableView/TableView.tsx (hunk 19 above)
Comment (Synicix) [reply src/Components/MainTableView/TableView.tsx:51]: fixed

File: src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx (hunk 20 above)
Comment (Synicix) [reply src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx:31]: fixed

File: src/Components/MainTableView/DataStorageClasses/PrimaryTableAttribute.tsx (hunk 12 above)
Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/PrimaryTableAttribute.tsx:63]: Same thing as the selectedTaableEntry

File: src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx (hunk 2 above)
Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx:73]: Fair enough

Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/SecondaryTableAttribute.tsx:73]: There, I updated the docs

File: src/Components/SideMenu/TableList.tsx [hunk 23]
```diff
@@ -128,53 +160,44 @@ class TableList extends React.Component<{
...
-        tableType = TableType.IMPORTED;
+  /**
+   * Given a list of parent table names, check if there is any partTables that belong to it, if so attach it then push to tableList, otherwise just push
+   * @param tableList 
+   * @param tableNames 
+   * @param tableType 
+   * @param partTableDict 
+   */
+  parseTableEntry(tableList: Array<ParentTableListEntry>, tableNames: Array<string>, tableType: TableType, partTableDict: Record<string, Array<PartTableListEntry>>) {
+    console.log(tableNames)
```
Comment (mahos) [inline src/Components/SideMenu/TableList.tsx:183]:
```suggestion
```
i spy a console.log

Comment (Synicix) [reply src/Components/SideMenu/TableList.tsx:183]: ah thanks, was trying to figure out why my quick rewrite broke the code

**Review by mahos — APPROVED** [review]

**Pull request merged**